lib/propolis-client/src/instance_spec.rs

```diff
@@ -168,6 +218,14 @@ pub enum StorageBackendKind {
     InMemory,
 }
 
+impl MigrationCompatible for StorageBackendKind {
+    fn is_migration_compatible(&self, other: &Self) -> bool {
+        // Two storage backends are compatible if they have the same kind,
+        // irrespective of their configurations.
+        std::mem::discriminant(self) == std::mem::discriminant(other)
```

Comment:
Does this preclude Crucible (or other backends) from changing in an incompatible way? Or would that be handled by introducing a new `StorageBackendKind`?